src/codeu/chat/server/JSON.java

```diff
@@ -385,25 +387,43 @@ public Model readFromFile (String file) throws IOException
                             jp.nextToken();
                             creation = Time.fromMs(jp.getLongValue());
                         }
+
                         if (jp.getText().equals("defaultType")) {
                             jp.nextToken();
                             defaultType = UserType.valueOf(jp.getText());
                         }
+                        if(jp.getText().equals("bots"))
+                        {
+                            jp.nextToken();
+                            while(jp.nextToken()!=JsonToken.END_OBJECT) {
```

Comment:
I'm wondering if this should be END_ARRAY instead of END_OBJECT? This depends on what the json looks like.